src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -218,12 +218,12 @@ numSnapRuns sl = V.sum $ V.map go1 sl
 type SnapLevels = V.Vector SnapLevel
 
 data SnapLevel = SnapLevel {
-    snapIncomingRuns :: !SnapMergingRun
+    snapIncoming     :: !SnapIncomingRun
   , snapResidentRuns :: !(V.Vector RunNumber)
   }
   deriving stock (Show, Eq)
 
-data SnapMergingRun =
+data SnapIncomingRun =
     SnapMergingRun !MergePolicyForLevel !NumRuns !NumEntries !UnspentCredits !MergeKnownCompleted !SnapMergingRunState
   | SnapSingleRun !RunNumber
   deriving stock (Show, Eq)
```

Comment:
Is it intentional that `SnapIncomingRun` represents the snapshot format of both `IncomingRun` and `MergingRun`?

Reply:
Well, `IncomingRun` also represents/includes `MergingRun`. `IncomingRun` and `SnapIncomingRun` are isomorphic. I just don't think adding another layer (factoring out into a `SnapMergingRun` type) is worth it.